src/components/scenes/SpendingLimitsScene.tsx

```diff
@@ -1,147 +1,111 @@
-  constructor(props: Props) {
-    super(props)
-    this.state = {
-      password: '',
-      transactionAmount: !zeroString(props.transactionSpendingLimit.amount.toString()) ? props.transactionSpendingLimit.amount.toString() : '',
-      transactionIsEnabled: props.transactionSpendingLimit.isEnabled
-    }
-  }
-
-  render() {
-    const { currencySymbol } = this.props
-    const { transactionAmount, transactionIsEnabled, password } = this.state
-
-    return (
-      <SceneWrapper background="legacy" hasHeader>
-        <KeyboardAwareScrollView contentContainerStyle={styles.scene}>
-          <OutlinedTextInput secureTextEntry autoFocus label={lstrings.enter_your_password} value={password} onChangeText={this.onPasswordChanged} />
+export const SpendingLimitsScene = (props: Props) => {
+  const { navigation } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  const dispatch = useDispatch()
 
-          <View style={styles.switchRow}>
-            <View style={styles.textBlock}>
-              <FormattedText style={styles.bodyText}>{lstrings.spending_limits_tx_title}</FormattedText>
-              <FormattedText style={styles.bodyText}>{lstrings.spending_limits_tx_description}</FormattedText>
-            </View>
-            <Switch onValueChange={this.onTransactionIsEnabledChanged} value={transactionIsEnabled} accessibilityHint={lstrings.toggle_button_hint} />
-          </View>
+  const currencySymbol = useSelector(state => getSymbolFromCurrency(state.ui.settings.defaultFiat))
+  const transactionSpendingLimit = useSelector(state => state.ui.settings.spendingLimits.transaction)
 
-          <OutlinedTextInput
-            disabled={!transactionIsEnabled}
-            value={transactionAmount}
-            onChangeText={this.onTransactionAmountChanged}
-            label={lstrings.spending_limits_tx_title}
-            autoCorrect={false}
-            autoFocus={false}
-            keyboardType="numeric"
-            prefix={currencySymbol}
-          />
+  const [password, setPassword] = React.useState('')
+  const [transactionAmount, setTransactionAmount] = React.useState(
+    !zeroString(transactionSpendingLimit.amount.toString()) ? transactionSpendingLimit.amount.toString() : ''
+  )
+  const [transactionIsEnabled, setTransactionIsEnabled] = React.useState(transactionSpendingLimit.isEnabled)
 
-          <View style={styles.spacer} />
+  const handleTransactionIsEnabledChanged = useHandler(() => setTransactionIsEnabled(!transactionIsEnabled))
+  const handleTransactionAmountChanged = useHandler((transactionAmount: string) => {
+    setTransactionAmount(transactionAmount)
+  })
+  const handlePasswordChanged = useHandler((password: string) => {
+    setPassword(password)
+  })
 
-          <PrimaryButton onPress={this.onSubmit}>
-            <PrimaryButton.Text>{lstrings.save}</PrimaryButton.Text>
-          </PrimaryButton>
-        </KeyboardAwareScrollView>
-      </SceneWrapper>
+  const handleSubmit = () => {
```

Comment:
Make this method `async`, and then go grab the contents out of `SpendingLimitsActions.ts` and move them into this method. They way it's factored now doesn't make sense, but since we have ported the scene to hooks, we can finally clean it up.